DQMOffline/LumiZCounting/plugins/ZCounting.cc

```diff
@@ -0,0 +1,472 @@
+        }
+      }
+      else if(itMu2->isStandAloneMuon()){
+        // category STA: probe is a STA muon
+        if(isProbeCentral){
+          h_mass_Sta_fail_central->Fill(iEvent.luminosityBlock(), vDilep.M());
+        }
+        else
+        {
+          h_mass_Sta_fail_forward->Fill(iEvent.luminosityBlock(), vDilep.M());
+        }
+      }
+      else if(itMu2->innerTrack()->hitPattern().trackerLayersWithMeasurement() >= 6 && itMu2->innerTrack()->hitPattern().numberOfValidPixelHits() >= 1){
+        // cateogry Trk: probe is a tracker track
+        if(isProbeCentral){
+          h_mass_Sta_fail_central->Fill(iEvent.luminosityBlock(), vDilep.M());
+        }
+        else
+        {
+          h_mass_Sta_fail_forward->Fill(iEvent.luminosityBlock(), vDilep.M());
+        }
+      }
+
+    }// End of probe loop over muons
+
+    // Probe loop over tracks, only for standalone efficiency calculation
+    for(reco::TrackCollection::const_iterator itTrk = trackCol->begin(); itTrk!=trackCol->end(); ++itTrk) {
+
+      // Check track is not a muon
+      bool isMuon = false;
+      for(reco::MuonCollection::const_iterator itMu = muonCol->begin(); itMu!=muonCol->end(); ++itMu) {
+        if(itMu->innerTrack().isNonnull() && itMu->innerTrack().get() == &(*itTrk)) {
+          isMuon = true;
+          break;
+        }
+      }
+      if(isMuon) continue;
+
+      float pt2  = itTrk->pt();
+      float eta2 = itTrk->eta();
+      float phi2 = itTrk->phi();
+      float q2   = itTrk->charge();
+
+      // Probe selection:  kinematic cuts and opposite charge requirement
+      if(pt2        < PtCutL2_)  continue;
+      if(fabs(eta2) > EtaCutL2_) continue;
+      if(q1 == q2)               continue;
+
+      vTrack.SetPtEtaPhiM(pt2, eta2, phi2, MUON_MASS);
+
+      TLorentzVector vDilep = vTag + vTrack;
+      if((vDilep.M() < MassMin_) || (vDilep.M() > MassMax_)) continue;
+
+      bool isTrackCentral = false;
+      if(fabs(eta2) > MUON_BOUND) isTrackCentral = true;
+
+      if(itTrk->hitPattern().trackerLayersWithMeasurement() >= 6 && itTrk->hitPattern().numberOfValidPixelHits() >= 1){
+        if(isTrackCentral) h_mass_Sta_fail_central->Fill(iEvent.luminosityBlock(), vDilep.M());
+        else               h_mass_Sta_fail_forward->Fill(iEvent.luminosityBlock(), vDilep.M());
+      }
+
+    }//End of probe loop over tracks
+
+  }//End of tag loop
+
+}
+
+//
+// -------------------------------------- endLuminosityBlock --------------------------------------------
+//
+void ZCounting::endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, edm::EventSetup const& eSetup)
+{
+  edm::LogInfo("ZCounting") <<  "ZCounting::endLuminosityBlock" << std::endl;
+}
+
+//
+// -------------------------------------- functions --------------------------------------------
+//
+void ZCounting::setTriggers()
+{
+  std::string cmssw_base_src = getenv("CMSSW_BASE"); cmssw_base_src+="/src/";
```

Comment:
This will most likely fail in any production environment. That is why it would be for the best to put those files into python configuration (PSet() and friends).